packages/devtools_app/lib/src/screens/performance/raster_metrics.dart

```diff
@@ -0,0 +1,355 @@
+        );
+
+  @override
+  double getValue(LayerSnapshot dataObject) =>
+      dataObject.percentRenderingTimeAsDouble;
+
+  @override
+  String getDisplayValue(LayerSnapshot dataObject) =>
+      dataObject.percentRenderingTimeDisplay;
+
+  @override
+  bool get numeric => true;
+}
+
+class LayerImage extends StatelessWidget {
+  const LayerImage({
+    Key? key,
+    required this.snapshot,
+    this.includeFullscreenButton = false,
+  }) : super(key: key);
+
+  final LayerSnapshot? snapshot;
+
+  final bool includeFullscreenButton;
+
+  double get _fullscreenButtonWidth => defaultButtonHeight + denseSpacing * 2;
+
+  static const _placeholderImageSize = 60.0;
+
+  @override
+  Widget build(BuildContext context) {
+    final color = Theme.of(context).focusColor;
+    final snapshot = this.snapshot;
+    if (snapshot == null) {
+      return const Icon(
+        Icons.image,
+        size: _placeholderImageSize,
+      );
+    }
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        SizedBox(
+          width:
+              includeFullscreenButton ? _fullscreenButtonWidth : defaultSpacing,
+        ),
+        Flexible(
+          child: Container(
+            color: color,
+            margin: const EdgeInsets.symmetric(horizontal: defaultSpacing),
+            child: Image.memory(snapshot.bytes),
+          ),
+        ),
+        includeFullscreenButton
+            ? Container(
+                padding: const EdgeInsets.only(bottom: denseSpacing),
+                alignment: Alignment.bottomRight,
+                child: IconButton(
+                  icon: Icon(
+                    Icons.fullscreen,
+                    size: defaultButtonHeight,
+                  ),
+                  onPressed: () {
+                    showDialog(
+                      context: context,
+                      builder: (context) => LayerImageDialog(
+                        snapshot: snapshot,
+                      ),
+                    );
+                  },
+                ),
+              )
+            : const SizedBox(width: defaultSpacing),
+      ],
+    );
+  }
+}
+
+class LayerImageDialog extends StatelessWidget {
+  const LayerImageDialog({Key? key, required this.snapshot}) : super(key: key);
+
+  final LayerSnapshot snapshot;
+
+  static const _padding = 300.0;
+
+  @override
+  Widget build(BuildContext context) {
+    final media = MediaQuery.of(context);
+    final mediaWidth = media.size.width;
+    final mediaHeight = media.size.height;
+    return DevToolsDialog(
+      includeDivider: false,
+      scrollable: false,
+      content: Container(
+        width: mediaWidth - _padding,
+        height: mediaHeight - _padding,
+        child: LayerImage(
+          snapshot: snapshot,
+        ),
+      ),
+      actions: [
+        DialogCloseButton(),
+      ],
+    );
+  }
+}
+
+class RasterMetricsController {
```

Comment:
This is just a curiosity - I know in other places we give the controller its own file. Is there a reason to include it here and not separate it out?

Reply:
I'll move it to a separate file in a follow up PR.